src/Pact/Analyze/Translate.hs

```diff
@@ -273,10 +272,10 @@ newtype TranslateM a
   deriving (Functor, Applicative, Monad, MonadReader TranslateEnv,
     MonadState TranslateState, MonadError TranslateFailure)
 
-instance MonadFail TranslateM where
-  fail s = do
-    info <- view envInfo
-    throwError (TranslateFailure info (MonadFailure s))
+failing :: String -> TranslateM a
+failing s = do
+  info <- view envInfo
+  throwError (TranslateFailure info (MonadFailure s))
```

Comment:
Instead of redefining this in multiple places, lets just add this to `Pact.Types.Util` and generalize to `MonadError m`

Reply:
This wasn't possible in the end.